tmt/steps/execute/scripts/tmt-report-result

```diff
@@ -12,62 +12,34 @@ die() { echo "$*" >&2; exit 2; }  # complain to STDERR and exit with error
-    echo "TESTNAME=$TESTNAME" >> $REPORT_RESULT_OUTPUTFILE
-    echo "TESTRESULT=$TESTRESULT" >> $REPORT_RESULT_OUTPUTFILE
-    echo "METRIC=$METRIC" >> $REPORT_RESULT_OUTPUTFILE
-}
-
-check_report_file () {
-    declare -A RESULTS_HIERARCHY=( ["FAIL"]=4 ["WARN"]=3 ["PASS"]=2 ["SKIP"]=1 )
-    # Create the output file in which to store the results.
-    REPORT_RESULT_OUTPUTFILE="$TMT_TEST_DATA/restraint-result"
-    if [ -e "$REPORT_RESULT_OUTPUTFILE" ]; then
-        # Check the existing output file.
-        # If new result is higher on the hierarchy
-        # or no result is contained in the file
-        # then overwrite the file with the new
-        # report contents.
-        fileResult=$(/usr/bin/grep "TESTRESULT" $REPORT_RESULT_OUTPUTFILE |
-        cut -d = -f 2)
-        if [ ! $fileResult ]; then
-            write_report_file;
-            exit 0
-        elif [ ! $TESTRESULT ]; then
-            exit 0
-        fi
-        fileHierarchy="${RESULTS_HIERARCHY[$fileResult]}"
-        # Compare the current result hierarchy against
-        # the hierarchy result retrieved from the file.
-        # Overwrite the file if the current hierachy is higher.
-        thisResultHierarchy="${RESULTS_HIERARCHY[$TESTRESULT]}"
-        if [ $thisResultHierarchy -gt $fileHierarchy ]; then
-            write_report_file
-            exit 0
-        fi
-        exit 0
-    else
-        # Create the output file with the report contents.
-        write_report_file
-        exit 0
+    if [ -n "$server" ]; then message="$message server=$server"; fi
+    if [ -n "$port" ]; then message="$message port=$port"; fi
+    if [ -n "$METRIC" ]; then message="$message metric=$METRIC"; fi
+    if [ -n "$disablePlugin" ]; then message="$message disablePlugin=$disablePlugin"; fi
+    echo "  - name: ${TESTNAME}" >> "$REPORT_RESULT_OUTPUTFILE"
+    echo "    fmf_id:" >> "$REPORT_RESULT_OUTPUTFILE"
+    echo "        name: ${TESTNAME}" >> "$REPORT_RESULT_OUTPUTFILE"
+    echo "    result: ${TESTRESULT}" >> "$REPORT_RESULT_OUTPUTFILE"
+    echo "    note: ${message}" >> "$REPORT_RESULT_OUTPUTFILE"
+    if [ -n "${outputFile}" ]; then
+        echo "    log:" >> "$REPORT_RESULT_OUTPUTFILE"
+        echo "      - ${outputFile}" >> "$REPORT_RESULT_OUTPUTFILE"
     fi
+    echo "    data-path: $TMT_TEST_DATA" >> "$REPORT_RESULT_OUTPUTFILE"
+    echo "">> "$REPORT_RESULT_OUTPUTFILE"
 }
 
-if [[ "$0" == *"rhts-report-result"* ]]; then
-    TESTNAME=$1
-    TESTRESULT=$2
-    outputFile=$3
-    METRIC=
-    if [ $# -gt 3 ];then
-        METRIC=$4
-    fi
-    check_report_file
-fi
+copy_outputfile_to_data_dir () {
+    filename=$(echo "$outputFile" | awk -F/ '{print $NF}')
+    # Replace slashes in test name with underscores.
+    fileprefix=$(echo "$TESTNAME" | tr / _ | tr ' ' _)
+    # Cut first char if underscore.
+    if [[ ${fileprefix:0:1} == '_' ]]; then fileprefix=$(echo "$fileprefix" | cut -c 2-); fi
```

Comment:
Would you please use `&&` instead of `if ... then ... fi`? e.g.

```bash
[[ ${fileprefix:0:1} == '_' ]] && fileprefix=$(echo "$fileprefix" | cut -c 2-)
```

Reply:
This suggestion was implemented. Resolving..